components/BookshelfItem.vue

```diff
@@ -85,22 +85,34 @@ const props = defineProps({
   },
 })
 
-const emit = defineEmits(['visible', 'open'])
+const emit = defineEmits(['visible', 'open', 'download'])
 
 const { t: $t } = useI18n()
 const accountStore = useAccountStore()
 const nftStore = useNFTStore()
 const metadataStore = useMetadataStore()
 const bookInfo = useBookInfo({ nftClassId: props.nftClassId })
+const { downloadBookFile } = useBookDownload()
+
+const { bookFileURLWithCORS } = useReader({
+  nftClassIdInput: props.nftClassId,
+})
 const bookCoverSrc = computed(() => getResizedImageURL(bookInfo.coverSrc.value, { size: 300 }))
 
 const isLargerScreen = useMediaQuery('(min-width: 1024px)')
 
+const arweaveLinkEndpoint = getArweaveLinkAPIEndpoint()
```

Comment:
I suggest to rename to `getEncryptedArweaveLinkAPIEndpoint()`
```suggestion
const encryptedArweaveLinkEndpoint = getEncryptedArweaveLinkAPIEndpoint()
```

Reply:
[🎨 Update Arweave endpoint and simplify parameter naming](https://github.com/likecoin/3ook-com/pull/47/commits/ba8def248f66aa90a8583bd483940f68fd008603)